src/test/ui/error-codes/E0523.rs

```diff
@@ -0,0 +1,20 @@
+/// Error is generated here:
+/// rust\compiler\rustc_metadata\src\locator.rs: 1015
+/// 
+/// ```
+/// CrateError::SymbolConflictsOthers(root_name) => struct_span_err!(
+///     sess,
+///     span,
+///     E0523,
+///     "found two different crates with name `{}` that are not distinguished \
+///      by differing \
+///      `-C metadata`. This will result in symbol conflicts between the two.",
+///     root_name,
+/// ),
+/// ```
+/// 
+
+// FIXME:  add code to reproduce the above error here
```

Comment:
I'm somewhat stuck here because I'm not sure how to create a small self-contained test that defines and pulls in two different crates with the same name. Is there a way to "stub" out a fake crate for use in a test like this?

(thanks in advance for the help -- I'm relatively new to rust and trying to learn a bit by working through some simple issues)

Reply:
So, in ui tests, there is a way to mimic extra crates. (I don't remember if 'auxillary' files are separate crates or not.

For the `rustc_error_codes` file, I think it's just going to be "faking" it.

Reply:
> there is a way to mimic extra crates

Would you be able to point me to an example or give me a keyword that I can search the repo for?

Reply:
https://rustc-dev-guide.rust-lang.org/tests/compiletest.html#building-auxiliary-crates